Xamarin.PropertyEditing/AssignableTypesResult.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Cadenza.Collections;
+
+namespace Xamarin.PropertyEditing
+{
+	public class AssignableTypesResult
+	{
+		public AssignableTypesResult (IReadOnlyCollection<ITypeInfo> assignableTypes)
+		{
+			if (assignableTypes == null)
+				throw new ArgumentNullException (nameof(assignableTypes));
+
+			AssignableTypes = assignableTypes;
+			SuggestedTypes = new ITypeInfo[0];
+		}
+
+		public AssignableTypesResult (IReadOnlyList<ITypeInfo> suggestedTypes, IReadOnlyCollection<ITypeInfo> assignableTypes)
+			: this (assignableTypes)
+		{
+			if (suggestedTypes == null)
+				throw new ArgumentNullException (nameof(suggestedTypes));
+
+			SuggestedTypes = suggestedTypes;
+		}
+
+		public IReadOnlyList<ITypeInfo> SuggestedTypes
+		{
+			get;
+		}
+
+		public IReadOnlyCollection<ITypeInfo> AssignableTypes
+		{
+			get;
+		}
+
+		internal IReadOnlyDictionary<IAssemblyInfo, ILookup<string, ITypeInfo>> GetTypeTree ()
+		{
+			var assemblies = new Dictionary<IAssemblyInfo, ILookup<string, ITypeInfo>> ();
```

Comment:
Any particular reason to not use a group by?

Reply:
Not off hand, it was already this, I just moved it.